src/App.tsx

```diff
@@ -1,33 +1,40 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from './App.module.scss';
-import Modal from "./components/Modal/Modal";
-import ContactsModalBody from "./components/ContactsModal/Body";
-import ContactsModalFooter from "./components/ContactsModal/Footer";
 import CustomButton from "./components/Button/Button";
+import { useLocation, useNavigate } from "react-router-dom";
+import ModalA from "./components/ContactsModal/ModalA";
+import ModalB from "./components/ContactsModal/ModalB";
 
 function App() {
-    const [showModalA, setShowModalA] = useState(window.location.hash === '#ModalA');
-    const [showModalB, setShowModalB] = useState(window.location.hash === '#ModalB');
+    const [showModalA, setShowModalA] = useState(false);
+    const [showModalB, setShowModalB] = useState(false);
+    const navigate = useNavigate();
+    const { hash } = useLocation();
     const [onlyEven, setOnlyEven] = useState(false);
 
+    useEffect(() => {
+        setShowModalA(hash === '#ModalA');
+        setShowModalB(hash === '#ModalB')
+    }, [hash])
+
     const openModalAHandler = () => {
         setShowModalA(true);
-        window.location.hash = '#ModalA';
+        navigate({ hash: '#ModalA' });
     }
 
     const openModalBHandler = () => {
         setShowModalB(true);
-        window.location.hash = '#ModalB';
+        navigate({ hash: '#ModalB' });
     }
 
     const closeModalAHandler = () => {
         setShowModalA(false)
-        window.location.hash = ''
+        navigate({ hash: '' });
```

Comment:
aici poti face un singur closeModal in care setezi hash la empty string. Acel useEffect va actualiza si state-urile